[VSEE-998] App-scanning 1.6 docs (#2597)
pivotal/docs-tap: scst-scan/integrate-app-scanning.hbs.md

> # Integration Guide

Reviewer: Would it be better as enabling app-scanning?  I'm trying to be consistent in the use of "bring your own scanner" but this isn't really that, this is really just the outline of how to enable it over Scan 1.0.

My thoughts right now are to merge it is as it is to get some of the important information out there, then iterate over it.  There are such substantial updates in this that I want to get it in and not block it by names of sections.

# Integration Guide

> This topic describes how to integrate Supply Chain Security Tools - Scan 2.0 with the Out of the Box Supply Chain.

Reviewer: ```suggestion
The default configuration for out of the box test + scan supply chain uses Supply Chain Security Tools - Scan 1.0. This topic describes how to integrate Supply Chain Security Tools - Scan 2.0 with the out of the box test + scan supply chain, as well as bring your own scanner by creating an ImageVulnerabilityScan template.
```

## <a id="prerequisites"></a> Prerequisites

> Before you can integrate Supply Chain Security Tools - Scan 2.0 with the Out of the Box Supply Chain:

Reviewer: ```suggestion
Before you can integrate Supply Chain Security Tools - Scan 2.0 with the out of the box supply chain:
```

> - Create an ImageVulnerabilityScan. You can either integrate your scanner or select from the provided samples:

Reviewer: ```suggestion
- Select an ImageVulnerabilityScan. You can either bring your own scanner by creating a custom ImageVulnerabilityScan template or select from the provided samples:
```

I think selecting an option is more appropriate than create right?  Since if you are using one of the sample templates you don't need to create it?

>   - [Customize your own ImageVulnerabilityScan](./ivs-create-your-own.hbs.md)

Reviewer: I'm trying to emphasize "bring your own scanner" and use consistent verbiage throughout the docs.

  - [ImageVulnerabilityScan samples](./ivs-custom-samples.hbs.md)

> - [Create a ClusterImageTemplate](./clusterimagetemplates.hbs.md). The previously created ImageVulnerabilityScan needs to be incorporated into a ClusterImageTemplate.

Reviewer: ```suggestion
- [Create a ClusterImageTemplate](./clusterimagetemplates.hbs.md). Incorporated the ImageVulnerabilityScan template into a ClusterImageTemplate.
```

## <a id="integration-with-supply-chain"></a> Integration with OOTB Supply Chain

1. After completing the prerequisites, update your `tap-values.yaml` file to specify the ClusterImageTemplate. For example:

    ```yaml
    ootb_supply_chain_testing_scanning:
      image_scanner_template_name: CLUSTERIMAGETEMPLATE
    ```

>     Where `CLUSTERIMAGETEMPLATE` is the ClusterImageTemplate with the embdedded ImageVulnerabilityScan using the scanner of your choice.

Reviewer: ```suggestion
    Where `CLUSTERIMAGETEMPLATE` is the name of the ClusterImageTemplate with the embedded ImageVulnerabilityScan using the scanner of your choice.
```

    ```console
    kubectl get imagevulnerabilityscans -n DEV-NAMESPACE -o wide

    NAME                 SCANRESULT                           SCANNEDIMAGE          SUCCEEDED   REASON
    generic-image-scan   registry/project/scan-results@digest nginx:latest@digest   True        Succeeded

    ```

### <a id="verifying-scanning-with-supply-chain-integration"></a> Verifying Scanning with Supply Chain Integration

1. Create a sample workload with a pre-built image by using the `tanzu apps workload create` command:

  ```console
  tanzu apps workload create WORKLOAD-NAME \
    --app APP-NAME \
    --type TYPE \
    --image IMAGE \
    --namespace DEV-NAMESPACE
  ```

  Where:

  - `WORKLOAD-NAME` is the name you choose for your workload.
  - `APP-NAME` is the name of your app.
  - `TYPE` is the type of your app.
  - `IMAGE` is the container image that contains the app you want to deploy.
  - `DEV-NAMESPACE` is the name of the developer namespace where scanning occurs.

  **Note**:
  - For more info on how to use the Tanzu CLI workload creation see [Create a Workload](../cli-plugins/apps/create-workload.hbs.md) documentation.

>   - Specifying the `--image` flag configures the workload to skip the source scanning step by using the specified pre-built image. Currently SCST - Scan 2.0 only performs image scanning. There are specific requirements for pre-built images. For more details see [Configure your workload to use a prebuilt image](../scc/pre-built-image.hbs.md)

Reviewer: ```suggestion
  - Specifying the `--image` flag configures the workload to skip the source scanning step if has been enabled by using the specified pre-built image. Currently SCST - Scan 2.0 only performs image scanning. There are specific requirements for pre-built images. For more details see [Configure your workload to use a prebuilt image](../scc/pre-built-image.hbs.md)
```
On the fence about removing this entirely, but I think the last part if important to capture.

Reviewer 2: Updated to be more specific on the supply chain being selected.
